lib/storage/src/content_manager/toc.rs

```diff
@@ -403,11 +451,24 @@ impl TableOfContent {
     async fn perform_collection_meta_op(
         &self,
         operation: CollectionMetaOperations,
+        shard_distribution_proposal: Option<ShardDistributionProposal>,
     ) -> Result<bool, StorageError> {
         match operation {
             CollectionMetaOperations::CreateCollection(operation) => {
-                self.create_collection(&operation.collection_name, operation.create_collection)
-                    .await
+                let collection_shard_distribution = match shard_distribution_proposal {
```

Comment:
why not make it a part of `CreateCollectionDistributed` struct, so it won't be possible to call `shard_distribution_proposal` with other types of collection opertions.

Reply:
What about this? https://github.com/qdrant/qdrant/pull/599/commits/5a04401d3a31d5cba533c2006ab9bdaa97ce38cf